texastribune/donations
# Slack updates (cancellations)

#### What's this PR do and why?

Introduces new rules for Slack notifications

- **Reduces noise:**
     - No more "payment_failed" in #bot-cancellations
- **Isolates circle:**
     - All circle issues go back into #circle-failures
     - This gives Irma and Morgan one place to look for failed circle payments


#### How should this be manually tested?

_At the time of writing this branch is hooked up to our staging site so you should be free to run these steps there:_
1. Create a few different new subscriptions.
     - Also fine if you have old ones, just be sure a circle rdo is attached to your account.
2. Hop into the portal with an account that can view as
     - Also fine to login as the identity you donated as if you donated as an existing portal account you control
3. Use @matthewdylan handy one-click cancel feature via the membership page
4. Confirm that the follow is correct:
     - Circle goes to **#bot-test** and reads something like this:
          - `Ashley Hebler's $84.00/monthly Circle donation was cancelled due to cancellation_requested (Member Portal)`
     - All others go to **#tech-test** and reads something like this:
          - `Slackity Slack's $349.00/yearly blast subscription was cancelled (Staff)`


😱 I didn't actually test the logic of "a circle rdo was cancelled bc of a failed payment" e.g. non cancellation-requested cancels. It seemed safe to assume those would still work, but let me know if I should bust out test clocks for that.

#### How should this change be communicated to end users?

I'll drop notes in `#circle-failures` and `#bot-cancellations` when this update is live

#### Are there any smells or added technical debt to note?

- It's more business rules to manage
- There might be better way to organize the code. Open to feedback there!

#### What are the relevant tickets?

https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwb2L2CHc4ZWqkC5/recyFUszW7KjLL4ng?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [x] Updated the documentation/wiki? *(Tried to document via comments. Maybe went overboard?)*

#### TODOs / next steps:

* [ ] *Add `SLACK_CIRCLE_NOTIFICATIONS` to heroku prod*
